sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/models/ScoringParameter.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.models;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a parameter value to be used in scoring functions (for example, referencePointParameter).
+ */
+public final class ScoringParameter {
+    private final ClientLogger logger = new ClientLogger(ScoringParameter.class);
+    private final String name;
+    private final List<String> values;
+
+    private static final String SEPARATOR = "-";
+    private static final String COMMA = ",";
+    private static final String SINGLE_QUOTE = "'";
+    /**
+     * Initializes a new instance of the ScoringParameter class with the given name and string values.
+     *
+     * @param name Name of the scoring parameter.
+     * @param values Values of the scoring parameter.
+     * @throws NullPointerException if {@code name} or {@code values} is null.
+     */
+    @JsonCreator
+    public ScoringParameter(@JsonProperty(value = "name") String name,
+        @JsonProperty(value = "values") List<String> values) {
+        Objects.requireNonNull(name);
+        Objects.requireNonNull(values);
+        this.name = name;
+        this.values = new ArrayList<>();
+        this.values.addAll(values);
+    }
+
+    /**
+     * Initializes a new instance of the ScoringParameter class with the given name and GeographyPoint value.
+     *
+     * @param name Name of the scoring parameter.
+     * @param value Value of the scoring parameter.
+     */
+    public ScoringParameter(String name, GeoPoint value) {
+        this(name, toLonLatStrings(value));
```

Comment:
Could this or `toLonLatStrings` be changed so the `NullPointerException` thrown from either of these methods instead of the other constructor when `value` is null.

Reply:
Put the null checking here.

Reply:
Put null checking in method at the first place.